### PR TITLE
Conformance Profiles: Badges for Reports

### LINKED ISSUE #2286: Conformance Profiles: initial display for reports
**What would you like to be added**:

A simple mechanism for showing that an implementation provides conformance profile reports in the implementations page, with links to the reports.

**Why this is needed**:

As per the graduation criteria of [Conformance Profiles (GEP 1709)](https://gateway-api.sigs.k8s.io/geps/gep-1709/).

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**

/kind cleanup
/kind documentation
/area conformance

**What this PR does / why we need it**:

This adds badges for implementations which are reporting conformance. The badges are placed under the organization name on the `implementations.md` page, and multiple badges can be listed if there are multiple projects. This helps us meet more of the graduation criteria for [GEP 1709](https://gateway-api.sigs.k8s.io/geps/gep-1709/) to graduate the conformance profiles work to standard.

For now, these badges are simple and manually managed so that we can just have _some_ display layer to start with. We expect that in the future we'll get tired of this and there will be more iterations to automate this process.

**Which issue(s) this PR fixes**:

Fixes #2286

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by shaneutt** [comment]
Not yet ready to review, I'm testing with the Netlify previews.

File: site-src/implementations.md [hunk 1]
```diff
@@ -318,6 +328,8 @@ The [LiteSpeed Ingress Controller](https://litespeedtech.com/products/litespeed-
 
 ### NGINX Kubernetes Gateway
 
+[![Conformance](https://img.shields.io/badge/Gateway%20API%20Conformance%20v0.7.1-NGinx Kubernetes Gateway-green)](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v0.7.1/nginxinc-nginx-kubernetes-gateway.yaml)
```
Comment (robscott) [inline site-src/implementations.md:331]:
```suggestion
[![Conformance](https://img.shields.io/badge/Gateway%20API%20Conformance%20v0.7.1-NGINX Kubernetes Gateway-green)](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v0.7.1/nginxinc-nginx-kubernetes-gateway.yaml)
```

File: geps/gep-1709.md [hunk 2]
```diff
@@ -573,9 +573,9 @@ point to the new data.
 > conformance test suite. When PRs are created the Gateway API repositories'
 > CI will run linting and validation against the reports.
 
-CI will provide [badges][bdg] to contributors at the end of the process which
-link to the implementations page for that specific implementation and can be
-easily added via markdown to Git repositories.
+Maintainers will provide [badges][bdg] to contributors at the end of the process
```
Comment (robscott) [inline geps/gep-1709.md:576]: non-blocking nit:
```suggestion
Maintainers will provide [badges][bdg] to implementations at the end of the process
```

**Review by robscott — COMMENTED** [review]
Thanks @shaneutt!

File: site-src/implementations.md [hunk 3]
```diff
@@ -318,6 +328,8 @@ The [LiteSpeed Ingress Controller](https://litespeedtech.com/products/litespeed-
 
 ### NGINX Kubernetes Gateway
 
+[![Conformance](https://img.shields.io/badge/Gateway%20API%20Conformance%20v0.7.1-NGINX Kubernetes Gateway-green)](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v0.7.1/nginxinc-nginx-kubernetes-gateway.yaml)
```
Comment (kate-osborn) [inline site-src/implementations.md:331]: @shaneutt this badge isn't rendering

Comment (shaneutt) [reply site-src/implementations.md:331]: Weird... I'm seeing it in the preview?

image

Comment (kate-osborn) [reply site-src/implementations.md:331]: Ah ok. It's not showing in the rich diff like the other badges, but if it's in the preview that's good 👍

**Pull request merged**